File: VirtualLab_Python/SimPla_Python/functions/tokamak.py
# -*- coding: utf-8 -*-
"""
Created on Mon Jun  2 14:40:29 2025

@author: ricca
"""

class tokamak:
    def __init__(self):
        self.machine = None
        self.R0 = None
        self.a = None
        self.wall = None
        self.grid = None
        self.separatrix = None
        self.config = {}

    def machine_upload(self, machine="Tokalab"):
        print(machine)

        if machine == "Tokalab":
            from tokamaks.geometry.Tokalab_Geometry import Tokalab_Geometry
            geo = Tokalab_Geometry()
        elif machine == "NewMachine":
            from tokamaks.geometry import NewMachine_Geometry
            geo = NewMachine_Geometry()
        else:
            raise ValueError(f"Unknown machine: {machine}")

        self.machine = machine
        self.R0 = geo.R0
        self.a = geo.a
        self.wall = geo.wall
        self.grid = geo.grid

    def scenario_upload(self,separatrix=1,Jt_method=1):
        if self.machine == "Tokalab":
            from tokamaks.equilibrium.Tokalab_Scenario import Tokalab_Scenario
            config = Tokalab_Scenario(separatrix,Jt_method)
        elif self.machine == "NewMachine":
            config = NewMachine_Scenario(separatrix,Jt_method)
        else:
            raise ValueError(f"Unknown machine: {self.machine}")

        self.config = config

    def kinetic_upload(self,kinetic_scenario=1):
        if self.machine == "Tokalab":
            from tokamaks.kinetic.Tokalab_Kinetic import Tokalab_Kinetic
            config = Tokalab_Kinetic(kinetic_scenario)
        elif self.machine == "NewMachine":
            config = NewMachine_Kinetic(kinetic_scenario)
        else:
            raise ValueError(f"Unknown machine: {self.machine}")

        self.config.kinetic = config.kinetic

File: VirtualLab_MatLab/SimPla_MATLAB/functions/tokamak.m

classdef tokamak

    properties
        machine % Tokamak name

        R0      % Major radius of the tokamak
        a       % Minor radius of the tokamak
        
        wall    % structure containing wall information
        
        grid    % grid of the geometry

        separatrix % here we store the target separatrix 

        config  % this structure contains various configuration
                % parameters for equilibrium 

    end

    methods
        % upload machine geometry
        function obj = machine_upload(obj,machine)
            
            if nargin < 2
                machine = "Tokalab";
            end

            disp(machine)

            if machine == "Tokalab"
                geo = Tokalab_Geometry();
            elseif machine == "NewMachine"
                geo = NewMachine_Geometry();
            end

            obj.machine = machine; % Store the machine name in the object
            obj.R0 = geo.R0;
            obj.a = geo.a;
            obj.wall = geo.wall;
            obj.grid = geo.grid;

        end

        % upload equilibrium configuration file
        function obj = scenario_upload(obj,separatrix,Jt_method)

            if nargin < 2
                separatrix = 1;
                Jt_method = 1;
            elseif nargin < 3
                Jt_method = 1;
            end

            machine = obj.machine;

            if machine == "Tokalab"
                config = Tokalab_Scenario(separatrix,Jt_method);
            elseif machine == "NewMachine"
                config = NewMachine_Scenario(separatrix,Jt_method);
            end

            % store parameters in the class
            obj.config = config;
            
        end

        % upload the kinetic configuration
        function obj = kinetic_upload(obj)
            
            machine = obj.machine;

            if machine == "Tokalab"
                config = Tokalab_Kinetic();
            elseif machine == "NewMachine"
                config = Tokalab_Kinetic();
            end

            % store parameters in the class
            obj.config.kinetic = config.kinetic;
            
        end

    end
end

            
